### DISCUSSION TITLE
Let's discuss about where to discuss (synchronously)

### DISCUSSION THREAD

**Discussion by rom1dep** [TriliumNext/Notes#7] [Polls]
## **TLDR**: The official TriliumNext synchronous discussion is happening on [Matrix](https://matrix.to/#/#triliumnext:matrix.org)
@TriliumNext/everyone please use the rooms in the matrix space for synchronous chat & coordination. If you would like a new room created for a specific github team, please let @meichthys know.

Hi!
Perhaps I'm beating a dead horse here, or maybe not, but I feel it's an important discussion to have, especially at this stage :)

This discussion can be split in parts:
1- should we care (what's at play),
2- a vote for one solution or another (if warranted),
3- a validation (over a certain period of time) that the solution fits the needs of most

This would be part 1.
In my opinion, self-hosted open-source projects should favour open-source and self-hostable communication systems, as to:

- guarantee inclusiveness: centralized protocols are easy targets for censors, which leaves Discord/Signal and many others blocked in [many countries](https://www.reddit.com/r/Maps/comments/zemjl9/countries_that_dont_allow_discord/), cutting-off a significant portion of the world population.
- be compatible: undocumented protocols or offering no backwards guarantees put their users at their mercy, practically controlling the access to the service and its user experience. Users with special accessibility needs, or depending on non-mainstream soft/hardware have no recourse.
- be sustainable: low interest rates are over and tech companies showed that they are no longer willing to trade freebies for extra growth (entering a phase that some would call "enshittification"). The pressure to monetize might cause large service providers to turn against their users, like seen many times before, with potential stakes regarding privacy (nobody's "too big to fail" on the internet, while the open-source offering only gets better over time).

Let me know what you think :)

**Comment by eliandoran** [comment]
Hi, @rom1dep .

Appreciate the fair take.

On my side I'm actually a quite ardent proponent of open-source. 99% of the applications I use are open-source, but some reason this hasn't yet translated to chatting applications.

I saw that @rauenzi had a Discord server already for it and I'm OK with using it. If the team thinks it's better to switch to something else, then I'm fine with it.

Theoretically GitHub is also not open, but at the same time we are using it because it happens to be convenient and it happens to be the place where everyone is.

Look forward to everyone's feedback.

**Comment by rodrigoalexandrecosta** [comment]
I mostly agree with @rom1dep regarding the reasons to use self-hosted FOSS, although I wouldn't oppose the use of @rauenzi 's Discord server for more synchronous communication until we have a more certain decision as a community on what tool to use.

I just want to point out that, while @eliandoran example about Github is correct, the underlying tool we're using is Git, meaning that even if Github decided to close our organization and ban us, we would have locally everything needed to go on to other solution, like Gitlab. It would be a headache and perhaps more than a nuisance, but it would be possible. 

But when we're dealing with close tools, we are somewhat pinned to the company behind the chatting software, and thus we may be affected by any kind of changes in their policies. Discord, for instance, AFAIK has no explicit policy saying that they can't use code or anything else that users post. Perhaps maintaining a self-hosted instance of e.g. Mattermost is more work, but at least we are not vulnerable to some weird bureaucratic policy change that just blocks or erase our chat history, including decisions, code snippets, etc.

That's my two cents.

**Reply by zerebos** [reply]
> even if Github decided to close our organization and ban us, we would have locally everything needed to go on to other solution, like Gitlab

If we are suddenly closed/banned we would not be retaining everything outside of the repo directly like issues, PRs, discussions, etc. That's a lot to lose in that case. GitHub is a closed-source Microsoft-owned tool that is nice and good to use. There's nothing inherently wrong with using closed-source especially when it works very well and benefits us.

**Reply by meichthys** [reply]
@rodrigoalexandrecosta has expressed my sentiments almost to the tee. Where possible (and not significantly inconvenient), i would prefer to stick with FOSS tools/solutions.

**Reply by eliandoran** [reply]
@rodrigoalexandrecosta , @meichthys , sounds good. Let's wait a bit until we get more people to vote and we can surely consider something else.

For me I'm quite happy with the discussions in the org for now.

**Reply by meichthys** [reply]
> If we are suddenly closed/banned we would not be retaining everything outside of the repo directly like issues, PRs, discussions, etc.

This would be a concern. If gitea[ implements issue sync ](https://github.com/go-gitea/gitea/pull/20311)on mirrored repositories, then, i'll probably run a mirror on it just in case (and for fun).

**Reply by rom1dep** [reply]
Yep, I'm not completely oblivious to the fact that I'm promoting decentralization on a platform that's the epitome of lock-in, bundling, and monopolistic bait-and-switch, and which contributed largely to almost killing every innovation in the space of version control system, CI/CD and many other aspects of project development.

That said, I'm very much keeping an eye on ForgeFed and its progress: https://codeberg.org/forgejo/forgejo/issues/59 , hoping it will prove to be good-enough to warrant a similar discussion here in the future ;)

**Reply by root-hal9000** [reply]
@meichthys I didn't know mirroring issues was on the roadmap for Gitea, that would be pretty sweet and make my isntance a lot more useful

**Reply by CobriMediaJulien** [reply]
Is the Discord active? I would love to join but the invitation i  found was expired.

**Reply by meichthys** [reply]
Active discussion is happening here https://app.element.io/#/room/#trilium:matrix.org

**Reply by zerebos** [reply]
The discord is still active yes, I was unaware that yet another chat was formed.

Here is a 7-day invite for the Discord https://discord.com/invite/pEgaTRVF

**Reply by meichthys** [reply]
@rom1dep and @rauenzi could we setup a mat4ix discord bridge so users can use both platforms? 
https://github.com/mautrix/discord

**Reply by zerebos** [reply]
@meichthys if it's feasible to do then of course, because I'd love to be able to communicate with those on the other platform. A quick look through the docs of that bridge makes it seem not so straightforward.

@Zocker1999NET I saw you offering to help create such bridges, do you have experience bridging matrix and Discord? or between matrix and Revolt?

**Reply by meichthys** [reply]
I asked the guys over on the Matrix chat if they had any experience with it and they also indicated that it is difficult and not very easy to maintain. 🤷‍♂️

**Reply by perfectra1n** [reply]
I have experience with Matrix and can give it a run as well.

**Reply by zerebos** [reply]
I'd be happy to work with you as well @perfectra1n. And ultimately if we can't get it working I'm alright conversing on GitHub with those only in Matrix. I saw another suggestion somewhere that we use email instead which isn't a bad idea either.

**Comment by Zocker1999NET** [comment]
TL;DR: IMO Matrix with bridging to other services should be the best compromise. I would like to assist in setting up that & can provide ressources to run such bridges if the majority would accept such solution.

I also agree with @rom1dep & @rodrigoalexandrecosta to prefer FOSS & open solutions over centralized properiatary solutions. Also, personally, I prefer solutions allowing / embracing federations and/or bridging. Personally, I prefer Matrix, but I understand the inconvience it brings with it.

I sadly have not much time to spare as of now, but otherwise would have the computational ressources and knowledge to aid in providing a bridging between Matrix and other services. Such a bridge should be possible between at least Discord, Telegram, and Matrix. A bridge to Mattermost seems to require a mostly dedicated Matrix server & a dedicated Mattermost server (at least when using [this connector](https://mattermost.com/marketplace/mattermost-matrix-connector/)), but also is technically possible.

I think such a solution is the best compromise as everyone use the tool which is most convinient to them (be it, e.g., Discord or a Matrix client). However, for transparency, some communities decided to disable such bridges again because it makes it harder for moderators to moderate (esp. on non-Matrix platforms where all bridged messages are issued by a single puppet account).

(For notice: I haven't contributed code to Trilium yet, just used it.)

**Reply by rom1dep** [reply]
as it happens, because Element bought gitter, we happen to already have a matrix room for Trilium :) 

https://app.gitter.im/#/room/#trilium-notes_Lobby:gitter.im

**Comment by zerebos** [comment]
I understand and agree in principle with the desire to use FOSS and standardized solutions for chatting/messaging. The vast majority of what I use is open source, and I even develop multiple open source projects. However for me, usability is king and none of the existing solutions are actually good to use. As Elian said, the open source chatting sphere is really lacking when compared to closed source solutions.

Hosting our own instance of some self-hosted FOSS solution to me seems like overkill for a group that hasn't even gotten off the ground yet. It adds overhead and maintenance for this group that's barely starting to form. Additionally, using such solutions will also discourage participation from active users due to the platforms being hard to use or understand or access or otherwise for the average user. For example the [official Trilium gitter](https://gitter.im/trilium-notes/Lobby) is fairly inactive with few members which is a poor participation rate for something as widely used as Trilium. The GitHub discussions which are slow and asynchronous were even used more than gitter.

In the end, if the group (once solidified who is a part of things) decides to move forward with one of the FOSS and/or self-hosted solutions for it then so be it I'll live. I'll probably just favor using GitHub more in that case personally.

**Reply by eliandoran** [reply]
@rauenzi , I was not even aware of the Trilium gitter. Interesting it has around 200 members.
Maybe someone from there would be interested in our org? :)

**Reply by zerebos** [reply]
Doesn't hurt to drop a message in there! The Gitter is linked at the top of the Trilium README.

**Reply by zadam** [reply]
I agree with this take. My advice would be to not split focus over very secondary things like communication platform, focus instead on the core of what you want to do. If Trilium Next takes off into stratosphere with many contributors, sure, migrate to self-hosted Matrix. For, now things like Discord are immediately available.

**Reply by rom1dep** [reply]
My idea starting the discussion like that was to avoid having everyone bring up their favourite platform and peer-pressure others, but since we are already here in this sub-thread…

> However for me, usability is king and none of the existing solutions are actually good to use. As Elian said, the open source chatting sphere is really lacking when compared to closed source solutions.

I respect your opinion, but did it occur to you that this could be the result of familiarity with one system over another? I've also heard terrible things about discord and how much fatigue its incurs and how low the signal to noise ratio generally is. That is to say that I don't believe that it is as clear cut as you make it sound, and all evidence points towards discord not even being as mainstream and dominant as you make it sound.

> Hosting our own instance of some self-hosted FOSS solution to me seems like overkill

If someone said anything about self-hosting, it wasn't me: all open/federated platforms that I know of have public instances that would welcome hosting such a project.

> For example the [official Trilium gitter](https://gitter.im/trilium-notes/Lobby) is fairly inactive with few members which is a poor participation rate for something as widely used as Trilium.

True, but you would have to demonstrate that this low activity is the result of an inadequacy of the solution (which I think is difficult, considering that it performs well for other much larger organizations) and that your alternative of choice would do measurably better. As it happens (anecdotal evidence), I found many answers to my programming questions from gitter channels that I wasn't even engaged in, just because the information was there, indexed and searchable, and this could be an excellent and factual reason as to why some open platforms are lower-paced: identical questions get asked (much) less often. Now compare this with discord, which purposely [locks its content behind a wall](https://news.ycombinator.com/item?id=35050858). I find this not just unnecessary, I think this is abusive, unfair and an insult to the goodwill and available time of its active users. I would never want to consider myself complicit dragging others into that, and in that regard discord is more divisive than most (open or closed, federated or centralized) alternatives.

**Comment by perfectra1n** [comment]
I share the same sentiments with everyone here - but Discord is just really, really good at what it offers. I also use Matrix, but it can be prohibitively complex sometimes (and you don't truly own your own data unless you run your own homeserver). IMO (I have a stupid opinion), to increase visibility into the "ecosystem" and to obtain more possible developer "mindshare", I think it would be a detriment if we weren't to heavily consider something like Discord. 

From what I've seen, tools that use Discord always gain more traction than tools that don't. Not sure if that's correlation or causation, but just my observation.

I'm also a Discord weeb and prefer my partner to message me there than text messaging, but just my 2c.

**Comment by root-hal9000** [comment]
So... are we actually using the Matrix one? I just joined, not seeing any messages

**Comment by CobriMediaJulien** [comment]
+ 1 for discord until we are better organized and then we can vote or something. But discord is nice to fast discuss and exchange expierences how to use trillium even better.

**Reply by meichthys** [reply]
Active discussion is currently happening on  matrix: https://app.element.io/#/room/#trilium:matrix.org
